src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -504,37 +538,60 @@ public static string GetMSBuildFilePathForPackageReferenceStyleProject(PackageSp
                 buildPropsGroup.RootName = MSBuildRestoreItemGroup.ImportGroup;
                 buildPropsGroup.Position = 2;
 
-                buildPropsGroup.Items.AddRange(sortedPackages.SelectMany(pkg =>
-                    pkg.Key.Build.WithExtension(PropsExtension)
-                        .Where(e => pkg.Value.Exists())
-                        .Select(e => pkg.Value.GetAbsolutePath(e)))
-                        .Select(path => GetPathWithMacros(path, repositoryRoot))
-                        .Select(GenerateImport));
+                foreach (var pkg in sortedPackages)
```

Comment:
I wonder if this is better off being a method to avoid bad bugs. We do the same logic (with a different comparer) for both targets and props